GTMAppAuth/Sources/KeychainStore/KeychainStore.swift

```diff
@@ -118,10 +129,20 @@ public final class KeychainStore: NSObject, AuthSessionStore {
   @objc(saveAuthSession:withItemName:error:)
   public func save(authSession: AuthSession, withItemName itemName: String) throws {
     let authSessionData = try authSessionData(fromAuthSession: authSession)
+
+    var maybeAccessibility: CFString? = kSecAttrAccessibleAfterFirstUnlockThisDeviceOnly
+    if #available(macOS 10.15, iOS 13.0, tvOS 13.0, watchOS 6.0, *) {
```

Comment:
`#if` is a compile-time option, `if #available` is evaluated at runtime. Wouldn't it be better to just have
```swift
if #available(macOS 10.15, *), !keychainAttributes.contains(.useDataProtectionKeychain) {
    // accessibility options on macOS only work when using the data protection keychain
    maybeAccessibility = nil
}
```

Reply:
I initially wanted to have a check like you have above, but it doesn't work as expected. As you point out, `if #available` is a runtime check, which means running tests on tvOS (for example) will fail due to `.useDataProtectionKeychain` not being available (or not being guaranteed to be available) during compilation.

Since we only want this logic to apply for certain versions of macOS, we need to do something like I have: check that we are running in an OS version where this attribute is available, check if we're on macOS, then check if the supplied attributes contains `.useDataProtectionKeychain`, and update the accessibility attribute accordingly.

Supplying the accessibility option on `KeychainAttribute` is a good idea, but doing so would change the previous/default behavior where we specify `kSecAttrAccessibleAfterFirstUnlockThisDeviceOnly` automatically - it would become an attribute that developers can control (by becoming a case on the `KeychainAttribute.Attribute` enum). 

We can consider something like this in the future, but I'd like to defer that to future consideration.